src/EVEMon.Common/QueryMonitor/CharacterDataQuerying.cs

```diff
@@ -231,12 +234,45 @@ internal void Dispose()
 
         #region Querying
         
+        /// <summary>
+        /// Handles the TimerTick event of the EveMonClient control.
+        /// Check if any character sheet related query monitors are still running, and trigger events if not.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void EveMonClient_TimerTick_CharacterSheetUpdated(object sender, EventArgs e)
+        {
+            // Check if all CharacterSheet related query monitors have completed
+            if (!m_characterQueryMonitors.Any(monitor =>
+                (ESIAPICharacterMethods.CharacterSheet.Equals(monitor.Method) || monitor.Method.HasParent(ESIAPICharacterMethods.CharacterSheet))
+                && (monitor.Status == QueryStatus.Updating)))
+            {
+                EveMonClient.TimerTick -= EveMonClient_TimerTick_CharacterSheetUpdated;
+                var target = m_ccpCharacter;
+                // Character may have been deleted since we queried
+                if (target != null)
+                {
+                    // Finally all done!
+                    EveMonClient.Notifications.InvalidateCharacterAPIError(target);
+                    EveMonClient.OnCharacterUpdated(target);
+                    EveMonClient.OnCharacterInfoUpdated(target);
+                    // Save character information locally
+                    var doc = Util.SerializeToXmlDocument(target.Export());
+                    LocalXmlCache.SaveAsync(target.Name, doc).ConfigureAwait(false);
+                }
+            }
+        }
+
         /// <summary>
         /// Processes the queried character's character sheet information.
         /// </summary>
         /// <param name="result"></param>
         private void OnCharacterSheetUpdated(EsiAPICharacterSheet result)
         {
+            // Subscribe timer tick to wait for the rest of the character sheet operations to finish
+            EveMonClient.TimerTick -= EveMonClient_TimerTick_CharacterSheetUpdated;
```

Comment:
If you are concerned about double subscription, you should use a Boolean member flag.